### PR TITLE
Add Unit of Work pattern

### PR DESCRIPTION
**Pull request by SolovyovDmitry**

### CODE REVIEW

File: src/Kernel.EFSupport/Provider/IRepository.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+[AutoInject]
+public interface IRepository<T> where T : class
+{
+  Task<IQueryable<T>> GetAllAsync();
+  Task<T> GetAsync(Guid id);
+  Task<Guid> CreateAsync(T entity);
+  Task<T> EditAsync(JsonPatchDocument<T> patch);
+  Task<T> UpdateAsync(T entity);
+  Task<bool> DeleteAsync(T entity);
```
Comment (ekaterinaryzhenkova) [inline src/Kernel.EFSupport/Provider/IRepository.cs:17]: Why not DeleteAsync(Guid id)?

Comment (SolovyovDmitry) [reply src/Kernel.EFSupport/Provider/IRepository.cs:17]: you will have to get it anyway so the idea is that you pass entity that was received from Get methods

File: src/Kernel.EFSupport/Provider/IRepository.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace LT.DigitalOffice.Kernel.EFSupport.Provider;
+
+[AutoInject]
+public interface IRepository<T> where T : class
+{
+  Task<IQueryable<T>> GetAllAsync();
+  Task<T> GetAsync(Guid id);
+  Task<Guid> CreateAsync(T entity);
+  Task<T> EditAsync(JsonPatchDocument<T> patch);
```
Comment (ekaterinaryzhenkova) [inline src/Kernel.EFSupport/Provider/IRepository.cs:15]: Why not return bool? Is it better to return entity?

File: src/Kernel.EFSupport/Provider/IRepository.cs (hunk 1 above)
Comment (ekaterinaryzhenkova) [reply src/Kernel.EFSupport/Provider/IRepository.cs:17]: got it

File: src/Kernel.EFSupport/Provider/IRepository.cs (hunk 2 above)
Comment (SolovyovDmitry) [reply src/Kernel.EFSupport/Provider/IRepository.cs:15]: it might be useful for frontend to return update entity after editing

Comment (ekaterinaryzhenkova) [reply src/Kernel.EFSupport/Provider/IRepository.cs:15]: got it

**Review by ekaterinaryzhenkova — APPROVED** [review]

**Pull request merged**